blog/2024-09-17-wasmcloud-postgres-capability-brings-pluggable-databases/index.mdx

```diff
@@ -0,0 +1,164 @@
+---
+title: "The wasmCloud Postgres capability brings pluggable databases for your apps"
+authors: [ericgregory]
+image: './images/wasmcloud-postgres-capability.webp'
+slug: 2024-09-17-wasmcloud-postgres-capability-brings-pluggable-databases
+date: "2024-09-17"
+description: "Learn how to get started with the Postgres capability in wasmCloud."
+---
+
+![The wasmCloud Postgres capability brings pluggable databases for your apps](./images/wasmcloud-postgres-capability.webp)
+
+With wasmCloud, you can build applications out of WebAssembly components and pluggable, reusable [**capabilities**](/docs/concepts/capabilities), so you can focus on the core logic of your application and abstract away the rest, all using open source standards.
+
+If you want to use a database system like **Postgres**, wasmCloud makes it easy to simply plug in the functionality in a reusable, language-agnostic way, without wasting time writing boilerplate code. In this post, we'll show you how to get started with the Postgres capability in wasmCloud.
+
+## Database capabilities speed up development
+
+Enterprise application teams need flexible database options, and for many, [PostgreSQL](https://www.postgresql.org/) is the tool of choice. But like many basic, non-functional requirements, teams can waste a lot of time getting to grips with libraries in this or that language and worrying about how to implement their database for a given project. 
+
+Capabilities abstract these worries away. In practice, every wasmCloud capability consists of two parts:
+
+* An **interface**: The standard contract for a piece of functionality, defined in [WebAssembly Interface Type (WIT)](/docs/concepts/interfaces#webassembly-interface-type-wit), defining an API for any language that uses the interface.
+* A **capability provider**:  The executable that plugs in to wasmCloud and delivers the functionality in question.
+
+In this case, you can use a [single API](https://github.com/wasmCloud/wasmCloud/tree/main/wit/postgres) for Postgres operations, regardless of your development language, and simply plug the [Postgres capability provider](https://github.com/wasmCloud/wasmCloud/tree/main/crates/provider-sqldb-postgres) into your application manifest when you deploy to wasmCloud. (Later on, if you need to migrate to another system, that's no problem&mdash;the `wasmcloud-postgres` interface can be used by any system that supports the WIT standard.)
+
```

Comment:
A nice addition might be to note that this *isn't* a one-size-fits-all SQL interface, it can take advantage of Postgres-specific features, like `RETURNING *` and DDL in queries with explicit `BEGIN`/`COMMIT` transactions ([`query-batch` interface](https://github.com/wasmCloud/wasmCloud/blob/main/wit/postgres/wit/query.wit#L26), if you want to link to something)